genesis/genesis.go

```diff
@@ -533,7 +559,10 @@ func splitAllocations(allocations []Allocation, numSplits int) [][]Allocation {
 		}
 	}
 
-	return append(allNodeAllocations, currentNodeAllocation)
+	if len(currentNodeAllocation) > 0 {
+		allNodeAllocations = append(allNodeAllocations, currentNodeAllocation)
+	}
+	return allNodeAllocations
```

Comment:
What does this modification do?

Reply:
not required for the fix and can be omitted. 
this modification ensures we don't return an array with an empty nested array when no locked amount. https://github.com/ava-labs/avalanchego/pull/3941/files/2842951c8860726232091bff268849ce0ff7117e#diff-2ce354664b078c8815c9e3644165471d1b47b509a4ce00763895428f2f1f384bR514-R515